lib/logstash/inputs/pipe.rb

```diff
@@ -56,25 +57,31 @@ def run(queue)
         end
         @pipe.close
         @pipe = nil
-      rescue LogStash::ShutdownSignal => e
-        break
       rescue Exception => e
         @logger.error("Exception while running command", :e => e, :backtrace => e.backtrace)
       end
 
       # Keep running the command forever.
-      sleep(10)
+      @sleep = true
```

Comment:
what is the purpose of this? Should we be using a synchronization mechanism instead? I'm not sure I like sleep+wakeup, but wait+notify is more common (a ConditionVariable) and I feel more confident in its thread safety.